src/components/dragLineLayer.ts

```diff
@@ -0,0 +1,170 @@
+///<reference path="../reference.ts" />
+
+module Plottable {
+
+export interface DragLineCallback<D> { (dragLineLayer: Components.DragLineLayer<D>): void; };
+
+export module Components {
+  export class DragLineLayer<D> extends GuideLineLayer<D> {
+    private _dragInteraction: Interactions.Drag;
+    private _detectionRadius = 3;
+    private _detectionEdge: d3.Selection<void>;
+    private _enabled = true;
+
+    private _dragStartCallbacks: Utils.CallbackSet<DragLineCallback<D>>;
+    private _dragCallbacks: Utils.CallbackSet<DragLineCallback<D>>;
+    private _dragEndCallbacks: Utils.CallbackSet<DragLineCallback<D>>;
+    private _disconnectInteractionCallbacks: () => void;
+
+    constructor(orientation: string) {
+      super(orientation);
+      this.addClass("drag-line-layer");
+      this.addClass("enabled");
+      if (this._isVertical()) {
+        this.addClass("vertical");
+      } else {
+        this.addClass("horizontal");
+      }
+
+      this._dragInteraction = new Plottable.Interactions.Drag();
+      this._dragInteraction.attachTo(this);
+
+      let grabbedLine = (p: Point) => {
+        return (this._isVertical() &&
+                 p.x >= this.pixelPosition() - this.detectionRadius() &&
+                 p.x <= this.pixelPosition() + this.detectionRadius()
+               ) ||
+               (
+                 !this._isVertical() &&
+                 p.y >= this.pixelPosition() - this.detectionRadius() &&
+                 p.y <= this.pixelPosition() + this.detectionRadius()
+               );
+      };
+
+      let dragging = false;
+      let dragStartCallback = (start: Point) => {
+        if (grabbedLine(start)) {
+          dragging = true;
+          this._dragStartCallbacks.callCallbacks(this);
+        }
+      };
+      this._dragInteraction.onDragStart(dragStartCallback);
+      let dragCallback = (start: Point, end: Point) => {
+        if (dragging) {
+          this._setPixelPositionWithoutChangingMode(this._isVertical() ? end.x : end.y);
+          this._dragCallbacks.callCallbacks(this);
+        }
+      };
+      this._dragInteraction.onDrag(dragCallback);
+      let dragEndCallback = (start: Point, end: Point) => {
+        if (dragging) {
+          dragging = false;
+          this._dragEndCallbacks.callCallbacks(this);
+        }
+      };
+      this._dragInteraction.onDragEnd(dragEndCallback);
```

Comment:
No, it makes sense because there's only one callback being added to the interaction. I could call it `interactionDragStartCallback`, if that made things clearer.